# Module 12 Challenge
## Deliverable 2: Scrape and Analyze Mars Weather Data

In [1]:
# Import relevant libraries
from splinter import Browser
from bs4 import BeautifulSoup as soup
import matplotlib.pyplot as plt
import pandas as pd

In [2]:
browser = Browser('chrome')

### Step 1: Visit the Website

Use automated browsing to visit the [Mars Temperature Data Site](https://static.bc-edx.com/data/web/mars_facts/temperature.html). Inspect the page to identify which elements to scrape.

   > **Hint** To identify which elements to scrape, you might want to inspect the page by using Chrome DevTools to discover whether the table contains usable classes.


In [3]:
# Visit the website
# https://static.bc-edx.com/data/web/mars_facts/temperature.html
url = ' https://static.bc-edx.com/data/web/mars_facts/temperature.html'
browser.visit(url)




### Step 2: Scrape the Table

Create a Beautiful Soup object and use it to scrape the data in the HTML table.

Note that this can also be achieved by using the Pandas `read_html` function. However, use Beautiful Soup here to continue sharpening your web scraping skills.

In [4]:
# Create a Beautiful Soup Object
html = browser.html
html_soup = soup(html, 'html.parser')

print(html_soup)

<html lang="en"><head>
<meta charset="utf-8"/>
<meta content="width=device-width, initial-scale=1" name="viewport"/>
<meta content="" name="description"/>
<title>Mars Temperature Data</title>
<link href="css/bootstrap.min.5.2.2.css" rel="stylesheet" type="text/css"/>
<link href="css/temp.css" rel="stylesheet" type="text/css"/>
</head>
<body>
<main>
<div class="container py-4">
<header class="pb-3 mb-4 border-bottom">
<a class="d-flex align-items-center text-dark text-decoration-none" href="/">
<span class="fs-4">Mission To Mars</span>
</a>
</header>
<div class="p-5 mb-4 bg-light rounded-3">
<div class="container-fluid py-5">
<h1 class="display-5 fw-bold">Mars Temperature Data</h1>
<p class="col-md-8 fs-4">This is Mars data.</p>
<table class="table">
<tbody><tr>
<th>id</th>
<th>terrestrial_date</th>
<th>sol</th>
<th>ls</th>
<th>month</th>
<th>min_temp</th>
<th>pressure</th>
</tr>
<tr class="data-row">
<td>2</td>
<td>2012-08-16</td>
<td>10</td>
<td>155</td>
<td>6</td>
<td>-75.0</td>
<td>

In [14]:
# Extract all rows of data
all_row = html_soup.find_all('tr', class_="data-row")
for tr in all_row:
    table_data = tr.find_all('td')
    print(table_data)
    for td in table_data:
        print(td.text)
        

[<td>2</td>, <td>2012-08-16</td>, <td>10</td>, <td>155</td>, <td>6</td>, <td>-75.0</td>, <td>739.0</td>]
2
2012-08-16
10
155
6
-75.0
739.0
[<td>13</td>, <td>2012-08-17</td>, <td>11</td>, <td>156</td>, <td>6</td>, <td>-76.0</td>, <td>740.0</td>]
13
2012-08-17
11
156
6
-76.0
740.0
[<td>24</td>, <td>2012-08-18</td>, <td>12</td>, <td>156</td>, <td>6</td>, <td>-76.0</td>, <td>741.0</td>]
24
2012-08-18
12
156
6
-76.0
741.0
[<td>35</td>, <td>2012-08-19</td>, <td>13</td>, <td>157</td>, <td>6</td>, <td>-74.0</td>, <td>732.0</td>]
35
2012-08-19
13
157
6
-74.0
732.0
[<td>46</td>, <td>2012-08-20</td>, <td>14</td>, <td>157</td>, <td>6</td>, <td>-74.0</td>, <td>740.0</td>]
46
2012-08-20
14
157
6
-74.0
740.0
[<td>57</td>, <td>2012-08-21</td>, <td>15</td>, <td>158</td>, <td>6</td>, <td>-78.0</td>, <td>740.0</td>]
57
2012-08-21
15
158
6
-78.0
740.0
[<td>68</td>, <td>2012-08-22</td>, <td>16</td>, <td>158</td>, <td>6</td>, <td>-77.0</td>, <td>740.0</td>]
68
2012-08-22
16
158
6
-77.0
740.0
[<td>79</td>, <

[<td>7</td>, <td>2012-11-20</td>, <td>104</td>, <td>210</td>, <td>8</td>, <td>-67.0</td>, <td>838.0</td>]
7
2012-11-20
104
210
8
-67.0
838.0
[<td>8</td>, <td>2012-11-21</td>, <td>105</td>, <td>211</td>, <td>8</td>, <td>-66.0</td>, <td>839.0</td>]
8
2012-11-21
105
211
8
-66.0
839.0
[<td>9</td>, <td>2012-11-23</td>, <td>106</td>, <td>211</td>, <td>8</td>, <td>-66.0</td>, <td>841.0</td>]
9
2012-11-23
106
211
8
-66.0
841.0
[<td>10</td>, <td>2012-11-24</td>, <td>107</td>, <td>212</td>, <td>8</td>, <td>-66.0</td>, <td>844.0</td>]
10
2012-11-24
107
212
8
-66.0
844.0
[<td>11</td>, <td>2012-11-25</td>, <td>108</td>, <td>212</td>, <td>8</td>, <td>-65.0</td>, <td>845.0</td>]
11
2012-11-25
108
212
8
-65.0
845.0
[<td>12</td>, <td>2012-11-26</td>, <td>109</td>, <td>213</td>, <td>8</td>, <td>-64.0</td>, <td>844.0</td>]
12
2012-11-26
109
213
8
-64.0
844.0
[<td>14</td>, <td>2012-11-27</td>, <td>110</td>, <td>214</td>, <td>8</td>, <td>-65.0</td>, <td>848.0</td>]
14
2012-11-27
110
214
8
-65.0
848.0
[<td>

247
9
-65.0
919.0
[<td>72</td>, <td>2013-01-20</td>, <td>163</td>, <td>248</td>, <td>9</td>, <td>-65.0</td>, <td>919.0</td>]
72
2013-01-20
163
248
9
-65.0
919.0
[<td>73</td>, <td>2013-01-21</td>, <td>164</td>, <td>248</td>, <td>9</td>, <td>-64.0</td>, <td>919.0</td>]
73
2013-01-21
164
248
9
-64.0
919.0
[<td>74</td>, <td>2013-01-22</td>, <td>165</td>, <td>249</td>, <td>9</td>, <td>-65.0</td>, <td>922.0</td>]
74
2013-01-22
165
249
9
-65.0
922.0
[<td>75</td>, <td>2013-01-23</td>, <td>166</td>, <td>250</td>, <td>9</td>, <td>-65.0</td>, <td>922.0</td>]
75
2013-01-23
166
250
9
-65.0
922.0
[<td>76</td>, <td>2013-01-24</td>, <td>167</td>, <td>250</td>, <td>9</td>, <td>-65.0</td>, <td>920.0</td>]
76
2013-01-24
167
250
9
-65.0
920.0
[<td>77</td>, <td>2013-01-25</td>, <td>168</td>, <td>251</td>, <td>9</td>, <td>-66.0</td>, <td>923.0</td>]
77
2013-01-25
168
251
9
-66.0
923.0
[<td>78</td>, <td>2013-01-26</td>, <td>169</td>, <td>252</td>, <td>9</td>, <td>-66.0</td>, <td>922.0</td>]
78
2013-01-26
169

871.0
[<td>154</td>, <td>2013-04-28</td>, <td>258</td>, <td>308</td>, <td>11</td>, <td>-71.0</td>, <td>869.0</td>]
154
2013-04-28
258
308
11
-71.0
869.0
[<td>155</td>, <td>2013-04-29</td>, <td>259</td>, <td>309</td>, <td>11</td>, <td>-72.0</td>, <td>871.0</td>]
155
2013-04-29
259
309
11
-72.0
871.0
[<td>157</td>, <td>2013-04-30</td>, <td>260</td>, <td>309</td>, <td>11</td>, <td>-70.0</td>, <td>875.0</td>]
157
2013-04-30
260
309
11
-70.0
875.0
[<td>158</td>, <td>2013-05-01</td>, <td>261</td>, <td>310</td>, <td>11</td>, <td>-70.0</td>, <td>868.0</td>]
158
2013-05-01
261
310
11
-70.0
868.0
[<td>161</td>, <td>2013-05-08</td>, <td>268</td>, <td>314</td>, <td>11</td>, <td>-70.0</td>, <td>864.0</td>]
161
2013-05-08
268
314
11
-70.0
864.0
[<td>162</td>, <td>2013-05-09</td>, <td>269</td>, <td>315</td>, <td>11</td>, <td>-72.0</td>, <td>866.0</td>]
162
2013-05-09
269
315
11
-72.0
866.0
[<td>164</td>, <td>2013-05-10</td>, <td>270</td>, <td>315</td>, <td>11</td>, <td>-67.0</td>, <td>863.0</td>]
164

[<td>300</td>, <td>2013-07-13</td>, <td>332</td>, <td>350</td>, <td>12</td>, <td>-76.0</td>, <td>849.0</td>]
300
2013-07-13
332
350
12
-76.0
849.0
[<td>299</td>, <td>2013-07-14</td>, <td>333</td>, <td>350</td>, <td>12</td>, <td>-75.0</td>, <td>849.0</td>]
299
2013-07-14
333
350
12
-75.0
849.0
[<td>301</td>, <td>2013-07-15</td>, <td>334</td>, <td>351</td>, <td>12</td>, <td>-74.0</td>, <td>852.0</td>]
301
2013-07-15
334
351
12
-74.0
852.0
[<td>302</td>, <td>2013-07-16</td>, <td>335</td>, <td>352</td>, <td>12</td>, <td>-76.0</td>, <td>852.0</td>]
302
2013-07-16
335
352
12
-76.0
852.0
[<td>305</td>, <td>2013-07-17</td>, <td>336</td>, <td>352</td>, <td>12</td>, <td>-76.0</td>, <td>852.0</td>]
305
2013-07-17
336
352
12
-76.0
852.0
[<td>304</td>, <td>2013-07-18</td>, <td>337</td>, <td>353</td>, <td>12</td>, <td>-75.0</td>, <td>853.0</td>]
304
2013-07-18
337
353
12
-75.0
853.0
[<td>306</td>, <td>2013-07-19</td>, <td>338</td>, <td>353</td>, <td>12</td>, <td>-78.0</td>, <td>848.0</td>]
306
2013-

24
1
-79.0
881.0
[<td>363</td>, <td>2013-09-21</td>, <td>400</td>, <td>24</td>, <td>1</td>, <td>-79.0</td>, <td>881.0</td>]
363
2013-09-21
400
24
1
-79.0
881.0
[<td>362</td>, <td>2013-09-22</td>, <td>401</td>, <td>25</td>, <td>1</td>, <td>-79.0</td>, <td>882.0</td>]
362
2013-09-22
401
25
1
-79.0
882.0
[<td>365</td>, <td>2013-09-23</td>, <td>402</td>, <td>25</td>, <td>1</td>, <td>-78.0</td>, <td>883.0</td>]
365
2013-09-23
402
25
1
-78.0
883.0
[<td>364</td>, <td>2013-09-24</td>, <td>403</td>, <td>25</td>, <td>1</td>, <td>-80.0</td>, <td>883.0</td>]
364
2013-09-24
403
25
1
-80.0
883.0
[<td>367</td>, <td>2013-09-25</td>, <td>404</td>, <td>26</td>, <td>1</td>, <td>-78.0</td>, <td>884.0</td>]
367
2013-09-25
404
26
1
-78.0
884.0
[<td>366</td>, <td>2013-09-26</td>, <td>405</td>, <td>26</td>, <td>1</td>, <td>-82.0</td>, <td>885.0</td>]
366
2013-09-26
405
26
1
-82.0
885.0
[<td>374</td>, <td>2013-09-27</td>, <td>406</td>, <td>27</td>, <td>1</td>, <td>-81.0</td>, <td>886.0</td>]
374
2013-09-27
406

2013-12-07
475
59
2
-85.0
906.0
[<td>418</td>, <td>2013-12-08</td>, <td>476</td>, <td>59</td>, <td>2</td>, <td>-85.0</td>, <td>906.0</td>]
418
2013-12-08
476
59
2
-85.0
906.0
[<td>421</td>, <td>2013-12-09</td>, <td>477</td>, <td>60</td>, <td>3</td>, <td>-85.0</td>, <td>907.0</td>]
421
2013-12-09
477
60
3
-85.0
907.0
[<td>422</td>, <td>2013-12-18</td>, <td>486</td>, <td>64</td>, <td>3</td>, <td>-84.0</td>, <td>904.0</td>]
422
2013-12-18
486
64
3
-84.0
904.0
[<td>473</td>, <td>2013-12-19</td>, <td>487</td>, <td>64</td>, <td>3</td>, <td>-85.0</td>, <td>904.0</td>]
473
2013-12-19
487
64
3
-85.0
904.0
[<td>480</td>, <td>2013-12-20</td>, <td>488</td>, <td>64</td>, <td>3</td>, <td>-87.0</td>, <td>903.0</td>]
480
2013-12-20
488
64
3
-87.0
903.0
[<td>423</td>, <td>2013-12-21</td>, <td>489</td>, <td>65</td>, <td>3</td>, <td>-85.0</td>, <td>904.0</td>]
423
2013-12-21
489
65
3
-85.0
904.0
[<td>434</td>, <td>2013-12-22</td>, <td>490</td>, <td>65</td>, <td>3</td>, <td>-87.0</td>, <td>903.0</td>]
434

89
3
-85.0
859.0
[<td>486</td>, <td>2014-02-15</td>, <td>543</td>, <td>89</td>, <td>3</td>, <td>-84.0</td>, <td>858.0</td>]
486
2014-02-15
543
89
3
-84.0
858.0
[<td>487</td>, <td>2014-02-16</td>, <td>544</td>, <td>90</td>, <td>4</td>, <td>-86.0</td>, <td>857.0</td>]
487
2014-02-16
544
90
4
-86.0
857.0
[<td>490</td>, <td>2014-02-17</td>, <td>545</td>, <td>90</td>, <td>4</td>, <td>-85.0</td>, <td>856.0</td>]
490
2014-02-17
545
90
4
-85.0
856.0
[<td>489</td>, <td>2014-02-18</td>, <td>546</td>, <td>91</td>, <td>4</td>, <td>-85.0</td>, <td>855.0</td>]
489
2014-02-18
546
91
4
-85.0
855.0
[<td>488</td>, <td>2014-02-19</td>, <td>547</td>, <td>91</td>, <td>4</td>, <td>-85.0</td>, <td>853.0</td>]
488
2014-02-19
547
91
4
-85.0
853.0
[<td>491</td>, <td>2014-02-20</td>, <td>548</td>, <td>91</td>, <td>4</td>, <td>-86.0</td>, <td>852.0</td>]
491
2014-02-20
548
91
4
-86.0
852.0
[<td>492</td>, <td>2014-02-21</td>, <td>549</td>, <td>92</td>, <td>4</td>, <td>-87.0</td>, <td>851.0</td>]
492
2014-02-21
549

5
-82.0
754.0
[<td>582</td>, <td>2014-05-21</td>, <td>636</td>, <td>133</td>, <td>5</td>, <td>-81.0</td>, <td>752.0</td>]
582
2014-05-21
636
133
5
-81.0
752.0
[<td>580</td>, <td>2014-05-22</td>, <td>637</td>, <td>134</td>, <td>5</td>, <td>-80.0</td>, <td>752.0</td>]
580
2014-05-22
637
134
5
-80.0
752.0
[<td>579</td>, <td>2014-05-23</td>, <td>638</td>, <td>134</td>, <td>5</td>, <td>-80.0</td>, <td>751.0</td>]
579
2014-05-23
638
134
5
-80.0
751.0
[<td>581</td>, <td>2014-05-24</td>, <td>639</td>, <td>135</td>, <td>5</td>, <td>-80.0</td>, <td>750.0</td>]
581
2014-05-24
639
135
5
-80.0
750.0
[<td>583</td>, <td>2014-05-25</td>, <td>640</td>, <td>135</td>, <td>5</td>, <td>-81.0</td>, <td>750.0</td>]
583
2014-05-25
640
135
5
-81.0
750.0
[<td>584</td>, <td>2014-05-26</td>, <td>641</td>, <td>136</td>, <td>5</td>, <td>-81.0</td>, <td>749.0</td>]
584
2014-05-26
641
136
5
-81.0
749.0
[<td>585</td>, <td>2014-05-27</td>, <td>642</td>, <td>136</td>, <td>5</td>, <td>-80.0</td>, <td>749.0</td>]
585
2014

7
-76.0
766.0
[<td>670</td>, <td>2014-08-24</td>, <td>728</td>, <td>183</td>, <td>7</td>, <td>-74.0</td>, <td>766.0</td>]
670
2014-08-24
728
183
7
-74.0
766.0
[<td>672</td>, <td>2014-08-25</td>, <td>729</td>, <td>183</td>, <td>7</td>, <td>-75.0</td>, <td>768.0</td>]
672
2014-08-25
729
183
7
-75.0
768.0
[<td>676</td>, <td>2014-08-26</td>, <td>730</td>, <td>184</td>, <td>7</td>, <td>-76.0</td>, <td>770.0</td>]
676
2014-08-26
730
184
7
-76.0
770.0
[<td>673</td>, <td>2014-08-27</td>, <td>731</td>, <td>185</td>, <td>7</td>, <td>-75.0</td>, <td>771.0</td>]
673
2014-08-27
731
185
7
-75.0
771.0
[<td>677</td>, <td>2014-08-28</td>, <td>732</td>, <td>185</td>, <td>7</td>, <td>-75.0</td>, <td>773.0</td>]
677
2014-08-28
732
185
7
-75.0
773.0
[<td>674</td>, <td>2014-08-29</td>, <td>733</td>, <td>186</td>, <td>7</td>, <td>-74.0</td>, <td>773.0</td>]
674
2014-08-29
733
186
7
-74.0
773.0
[<td>675</td>, <td>2014-08-30</td>, <td>734</td>, <td>186</td>, <td>7</td>, <td>-73.0</td>, <td>776.0</td>]
675
2014

[<td>736</td>, <td>2014-10-29</td>, <td>793</td>, <td>223</td>, <td>8</td>, <td>-71.0</td>, <td>877.0</td>]
736
2014-10-29
793
223
8
-71.0
877.0
[<td>739</td>, <td>2014-10-30</td>, <td>794</td>, <td>223</td>, <td>8</td>, <td>-68.0</td>, <td>874.0</td>]
739
2014-10-30
794
223
8
-68.0
874.0
[<td>738</td>, <td>2014-10-31</td>, <td>795</td>, <td>224</td>, <td>8</td>, <td>-68.0</td>, <td>875.0</td>]
738
2014-10-31
795
224
8
-68.0
875.0
[<td>737</td>, <td>2014-11-01</td>, <td>796</td>, <td>225</td>, <td>8</td>, <td>-65.0</td>, <td>877.0</td>]
737
2014-11-01
796
225
8
-65.0
877.0
[<td>740</td>, <td>2014-11-03</td>, <td>797</td>, <td>225</td>, <td>8</td>, <td>-65.0</td>, <td>879.0</td>]
740
2014-11-03
797
225
8
-65.0
879.0
[<td>741</td>, <td>2014-11-04</td>, <td>798</td>, <td>226</td>, <td>8</td>, <td>-67.0</td>, <td>883.0</td>]
741
2014-11-04
798
226
8
-67.0
883.0
[<td>743</td>, <td>2014-11-05</td>, <td>799</td>, <td>227</td>, <td>8</td>, <td>-70.0</td>, <td>884.0</td>]
743
2014-11-05
799
227

[<td>795</td>, <td>2015-01-02</td>, <td>856</td>, <td>264</td>, <td>9</td>, <td>-68.0</td>, <td>912.0</td>]
795
2015-01-02
856
264
9
-68.0
912.0
[<td>801</td>, <td>2015-01-03</td>, <td>857</td>, <td>264</td>, <td>9</td>, <td>-76.0</td>, <td>911.0</td>]
801
2015-01-03
857
264
9
-76.0
911.0
[<td>800</td>, <td>2015-01-04</td>, <td>858</td>, <td>265</td>, <td>9</td>, <td>-73.0</td>, <td>913.0</td>]
800
2015-01-04
858
265
9
-73.0
913.0
[<td>802</td>, <td>2015-01-05</td>, <td>859</td>, <td>266</td>, <td>9</td>, <td>-72.0</td>, <td>915.0</td>]
802
2015-01-05
859
266
9
-72.0
915.0
[<td>803</td>, <td>2015-01-06</td>, <td>860</td>, <td>266</td>, <td>9</td>, <td>-69.0</td>, <td>916.0</td>]
803
2015-01-06
860
266
9
-69.0
916.0
[<td>804</td>, <td>2015-01-07</td>, <td>861</td>, <td>267</td>, <td>9</td>, <td>-70.0</td>, <td>917.0</td>]
804
2015-01-07
861
267
9
-70.0
917.0
[<td>807</td>, <td>2015-01-08</td>, <td>862</td>, <td>268</td>, <td>9</td>, <td>-70.0</td>, <td>917.0</td>]
807
2015-01-08
862
268

311
11
-71.0
859.0
[<td>869</td>, <td>2015-03-22</td>, <td>933</td>, <td>312</td>, <td>11</td>, <td>-71.0</td>, <td>863.0</td>]
869
2015-03-22
933
312
11
-71.0
863.0
[<td>870</td>, <td>2015-03-23</td>, <td>934</td>, <td>313</td>, <td>11</td>, <td>-73.0</td>, <td>858.0</td>]
870
2015-03-23
934
313
11
-73.0
858.0
[<td>872</td>, <td>2015-03-24</td>, <td>935</td>, <td>313</td>, <td>11</td>, <td>-75.0</td>, <td>862.0</td>]
872
2015-03-24
935
313
11
-75.0
862.0
[<td>871</td>, <td>2015-03-25</td>, <td>936</td>, <td>314</td>, <td>11</td>, <td>-72.0</td>, <td>865.0</td>]
871
2015-03-25
936
314
11
-72.0
865.0
[<td>874</td>, <td>2015-03-26</td>, <td>937</td>, <td>314</td>, <td>11</td>, <td>-73.0</td>, <td>863.0</td>]
874
2015-03-26
937
314
11
-73.0
863.0
[<td>876</td>, <td>2015-03-29</td>, <td>940</td>, <td>316</td>, <td>11</td>, <td>-73.0</td>, <td>861.0</td>]
876
2015-03-29
940
316
11
-73.0
861.0
[<td>875</td>, <td>2015-03-30</td>, <td>941</td>, <td>317</td>, <td>11</td>, <td>-74.0</td>, <td>85

[<td>928</td>, <td>2015-05-26</td>, <td>996</td>, <td>348</td>, <td>12</td>, <td>-72.0</td>, <td>842.0</td>]
928
2015-05-26
996
348
12
-72.0
842.0
[<td>930</td>, <td>2015-05-27</td>, <td>997</td>, <td>348</td>, <td>12</td>, <td>-74.0</td>, <td>843.0</td>]
930
2015-05-27
997
348
12
-74.0
843.0
[<td>931</td>, <td>2015-05-28</td>, <td>998</td>, <td>349</td>, <td>12</td>, <td>-73.0</td>, <td>842.0</td>]
931
2015-05-28
998
349
12
-73.0
842.0
[<td>932</td>, <td>2015-05-30</td>, <td>1000</td>, <td>350</td>, <td>12</td>, <td>-74.0</td>, <td>841.0</td>]
932
2015-05-30
1000
350
12
-74.0
841.0
[<td>933</td>, <td>2015-05-31</td>, <td>1001</td>, <td>350</td>, <td>12</td>, <td>-79.0</td>, <td>843.0</td>]
933
2015-05-31
1001
350
12
-79.0
843.0
[<td>936</td>, <td>2015-06-01</td>, <td>1002</td>, <td>351</td>, <td>12</td>, <td>-77.0</td>, <td>844.0</td>]
936
2015-06-01
1002
351
12
-77.0
844.0
[<td>934</td>, <td>2015-06-02</td>, <td>1003</td>, <td>351</td>, <td>12</td>, <td>-74.0</td>, <td>843.0</td>]
93

[<td>992</td>, <td>2015-07-31</td>, <td>1060</td>, <td>20</td>, <td>1</td>, <td>-78.0</td>, <td>870.0</td>]
992
2015-07-31
1060
20
1
-78.0
870.0
[<td>993</td>, <td>2015-08-01</td>, <td>1061</td>, <td>20</td>, <td>1</td>, <td>-77.0</td>, <td>871.0</td>]
993
2015-08-01
1061
20
1
-77.0
871.0
[<td>991</td>, <td>2015-08-02</td>, <td>1062</td>, <td>21</td>, <td>1</td>, <td>-77.0</td>, <td>871.0</td>]
991
2015-08-02
1062
21
1
-77.0
871.0
[<td>994</td>, <td>2015-08-03</td>, <td>1063</td>, <td>21</td>, <td>1</td>, <td>-79.0</td>, <td>871.0</td>]
994
2015-08-03
1063
21
1
-79.0
871.0
[<td>996</td>, <td>2015-08-04</td>, <td>1064</td>, <td>22</td>, <td>1</td>, <td>-77.0</td>, <td>871.0</td>]
996
2015-08-04
1064
22
1
-77.0
871.0
[<td>995</td>, <td>2015-08-05</td>, <td>1065</td>, <td>22</td>, <td>1</td>, <td>-78.0</td>, <td>873.0</td>]
995
2015-08-05
1065
22
1
-78.0
873.0
[<td>997</td>, <td>2015-08-06</td>, <td>1066</td>, <td>23</td>, <td>1</td>, <td>-78.0</td>, <td>875.0</td>]
997
2015-08-06
1066
23

2015-10-15
1134
54
2
-80.0
902.0
[<td>1069</td>, <td>2015-10-16</td>, <td>1135</td>, <td>55</td>, <td>2</td>, <td>-81.0</td>, <td>901.0</td>]
1069
2015-10-16
1135
55
2
-81.0
901.0
[<td>1066</td>, <td>2015-10-17</td>, <td>1136</td>, <td>55</td>, <td>2</td>, <td>-82.0</td>, <td>901.0</td>]
1066
2015-10-17
1136
55
2
-82.0
901.0
[<td>1068</td>, <td>2015-10-18</td>, <td>1137</td>, <td>56</td>, <td>2</td>, <td>-81.0</td>, <td>901.0</td>]
1068
2015-10-18
1137
56
2
-81.0
901.0
[<td>1067</td>, <td>2015-10-19</td>, <td>1138</td>, <td>56</td>, <td>2</td>, <td>-82.0</td>, <td>901.0</td>]
1067
2015-10-19
1138
56
2
-82.0
901.0
[<td>1070</td>, <td>2015-10-20</td>, <td>1139</td>, <td>57</td>, <td>2</td>, <td>-80.0</td>, <td>901.0</td>]
1070
2015-10-20
1139
57
2
-80.0
901.0
[<td>1071</td>, <td>2015-10-21</td>, <td>1140</td>, <td>57</td>, <td>2</td>, <td>-81.0</td>, <td>901.0</td>]
1071
2015-10-21
1140
57
2
-81.0
901.0
[<td>1072</td>, <td>2015-10-22</td>, <td>1141</td>, <td>57</td>, <td>2</td>, <td>-82.

[<td>1159</td>, <td>2016-01-19</td>, <td>1228</td>, <td>97</td>, <td>4</td>, <td>-88.0</td>, <td>832.0</td>]
1159
2016-01-19
1228
97
4
-88.0
832.0
[<td>1160</td>, <td>2016-01-20</td>, <td>1229</td>, <td>97</td>, <td>4</td>, <td>-88.0</td>, <td>831.0</td>]
1160
2016-01-20
1229
97
4
-88.0
831.0
[<td>1161</td>, <td>2016-01-21</td>, <td>1230</td>, <td>98</td>, <td>4</td>, <td>-88.0</td>, <td>830.0</td>]
1161
2016-01-21
1230
98
4
-88.0
830.0
[<td>1162</td>, <td>2016-01-22</td>, <td>1231</td>, <td>98</td>, <td>4</td>, <td>-88.0</td>, <td>828.0</td>]
1162
2016-01-22
1231
98
4
-88.0
828.0
[<td>1163</td>, <td>2016-01-23</td>, <td>1232</td>, <td>99</td>, <td>4</td>, <td>-88.0</td>, <td>827.0</td>]
1163
2016-01-23
1232
99
4
-88.0
827.0
[<td>1164</td>, <td>2016-01-24</td>, <td>1233</td>, <td>99</td>, <td>4</td>, <td>-88.0</td>, <td>826.0</td>]
1164
2016-01-24
1233
99
4
-88.0
826.0
[<td>1166</td>, <td>2016-01-26</td>, <td>1234</td>, <td>99</td>, <td>4</td>, <td>-88.0</td>, <td>824.0</td>]
1166
2016

1215
2016-03-19
1286
124
5
-80.0
765.0
[<td>1218</td>, <td>2016-03-20</td>, <td>1287</td>, <td>124</td>, <td>5</td>, <td>-80.0</td>, <td>764.0</td>]
1218
2016-03-20
1287
124
5
-80.0
764.0
[<td>1219</td>, <td>2016-03-21</td>, <td>1288</td>, <td>125</td>, <td>5</td>, <td>-80.0</td>, <td>763.0</td>]
1219
2016-03-21
1288
125
5
-80.0
763.0
[<td>1220</td>, <td>2016-03-24</td>, <td>1291</td>, <td>126</td>, <td>5</td>, <td>-80.0</td>, <td>760.0</td>]
1220
2016-03-24
1291
126
5
-80.0
760.0
[<td>1222</td>, <td>2016-03-25</td>, <td>1292</td>, <td>127</td>, <td>5</td>, <td>-81.0</td>, <td>760.0</td>]
1222
2016-03-25
1292
127
5
-81.0
760.0
[<td>1221</td>, <td>2016-03-26</td>, <td>1293</td>, <td>127</td>, <td>5</td>, <td>-81.0</td>, <td>758.0</td>]
1221
2016-03-26
1293
127
5
-81.0
758.0
[<td>1223</td>, <td>2016-03-27</td>, <td>1294</td>, <td>128</td>, <td>5</td>, <td>-79.0</td>, <td>758.0</td>]
1223
2016-03-27
1294
128
5
-79.0
758.0
[<td>1224</td>, <td>2016-03-28</td>, <td>1295</td>, <td>128</td>, <

1369
167
6
-73.0
742.0
[<td>1298</td>, <td>2016-06-13</td>, <td>1370</td>, <td>168</td>, <td>6</td>, <td>-74.0</td>, <td>744.0</td>]
1298
2016-06-13
1370
168
6
-74.0
744.0
[<td>1301</td>, <td>2016-06-14</td>, <td>1371</td>, <td>168</td>, <td>6</td>, <td>-75.0</td>, <td>746.0</td>]
1301
2016-06-14
1371
168
6
-75.0
746.0
[<td>1300</td>, <td>2016-06-15</td>, <td>1372</td>, <td>169</td>, <td>6</td>, <td>-74.0</td>, <td>745.0</td>]
1300
2016-06-15
1372
169
6
-74.0
745.0
[<td>1303</td>, <td>2016-06-16</td>, <td>1373</td>, <td>169</td>, <td>6</td>, <td>-74.0</td>, <td>746.0</td>]
1303
2016-06-16
1373
169
6
-74.0
746.0
[<td>1302</td>, <td>2016-06-17</td>, <td>1374</td>, <td>170</td>, <td>6</td>, <td>-75.0</td>, <td>748.0</td>]
1302
2016-06-17
1374
170
6
-75.0
748.0
[<td>1304</td>, <td>2016-06-18</td>, <td>1375</td>, <td>170</td>, <td>6</td>, <td>-73.0</td>, <td>749.0</td>]
1304
2016-06-18
1375
170
6
-73.0
749.0
[<td>1305</td>, <td>2016-06-19</td>, <td>1376</td>, <td>171</td>, <td>6</td>, <td>-

1434
205
7
-70.0
818.0
[<td>1355</td>, <td>2016-08-19</td>, <td>1435</td>, <td>206</td>, <td>7</td>, <td>-71.0</td>, <td>823.0</td>]
1355
2016-08-19
1435
206
7
-71.0
823.0
[<td>1354</td>, <td>2016-08-20</td>, <td>1436</td>, <td>207</td>, <td>7</td>, <td>-72.0</td>, <td>825.0</td>]
1354
2016-08-20
1436
207
7
-72.0
825.0
[<td>1357</td>, <td>2016-08-21</td>, <td>1437</td>, <td>207</td>, <td>7</td>, <td>-71.0</td>, <td>821.0</td>]
1357
2016-08-21
1437
207
7
-71.0
821.0
[<td>1358</td>, <td>2016-08-22</td>, <td>1438</td>, <td>208</td>, <td>7</td>, <td>-68.0</td>, <td>823.0</td>]
1358
2016-08-22
1438
208
7
-68.0
823.0
[<td>1360</td>, <td>2016-08-23</td>, <td>1439</td>, <td>208</td>, <td>7</td>, <td>-71.0</td>, <td>828.0</td>]
1360
2016-08-23
1439
208
7
-71.0
828.0
[<td>1359</td>, <td>2016-08-24</td>, <td>1440</td>, <td>209</td>, <td>7</td>, <td>-70.0</td>, <td>828.0</td>]
1359
2016-08-24
1440
209
7
-70.0
828.0
[<td>1361</td>, <td>2016-08-25</td>, <td>1441</td>, <td>210</td>, <td>8</td>, <td>-

[<td>1415</td>, <td>2016-10-21</td>, <td>1496</td>, <td>245</td>, <td>9</td>, <td>-72.0</td>, <td>906.0</td>]
1415
2016-10-21
1496
245
9
-72.0
906.0
[<td>1417</td>, <td>2016-10-22</td>, <td>1497</td>, <td>246</td>, <td>9</td>, <td>-68.0</td>, <td>909.0</td>]
1417
2016-10-22
1497
246
9
-68.0
909.0
[<td>1418</td>, <td>2016-10-23</td>, <td>1498</td>, <td>246</td>, <td>9</td>, <td>-68.0</td>, <td>907.0</td>]
1418
2016-10-23
1498
246
9
-68.0
907.0
[<td>1419</td>, <td>2016-10-24</td>, <td>1499</td>, <td>247</td>, <td>9</td>, <td>-69.0</td>, <td>903.0</td>]
1419
2016-10-24
1499
247
9
-69.0
903.0
[<td>1420</td>, <td>2016-10-25</td>, <td>1500</td>, <td>248</td>, <td>9</td>, <td>-70.0</td>, <td>903.0</td>]
1420
2016-10-25
1500
248
9
-70.0
903.0
[<td>1421</td>, <td>2016-10-26</td>, <td>1501</td>, <td>248</td>, <td>9</td>, <td>-69.0</td>, <td>903.0</td>]
1421
2016-10-26
1501
248
9
-69.0
903.0
[<td>1422</td>, <td>2016-10-27</td>, <td>1502</td>, <td>249</td>, <td>9</td>, <td>-69.0</td>, <td>907.0</t

1564
289
10
-70.0
877.0
[<td>1480</td>, <td>2016-12-31</td>, <td>1565</td>, <td>290</td>, <td>10</td>, <td>-72.0</td>, <td>874.0</td>]
1480
2016-12-31
1565
290
10
-72.0
874.0
[<td>1487</td>, <td>2017-01-01</td>, <td>1566</td>, <td>290</td>, <td>10</td>, <td>-76.0</td>, <td>876.0</td>]
1487
2017-01-01
1566
290
10
-76.0
876.0
[<td>1486</td>, <td>2017-01-02</td>, <td>1567</td>, <td>291</td>, <td>10</td>, <td>-70.0</td>, <td>872.0</td>]
1486
2017-01-02
1567
291
10
-70.0
872.0
[<td>1488</td>, <td>2017-01-03</td>, <td>1568</td>, <td>291</td>, <td>10</td>, <td>-71.0</td>, <td>876.0</td>]
1488
2017-01-03
1568
291
10
-71.0
876.0
[<td>1492</td>, <td>2017-01-04</td>, <td>1569</td>, <td>292</td>, <td>10</td>, <td>-70.0</td>, <td>873.0</td>]
1492
2017-01-04
1569
292
10
-70.0
873.0
[<td>1491</td>, <td>2017-01-05</td>, <td>1570</td>, <td>293</td>, <td>10</td>, <td>-71.0</td>, <td>873.0</td>]
1491
2017-01-05
1570
293
10
-71.0
873.0
[<td>1490</td>, <td>2017-01-06</td>, <td>1571</td>, <td>293</td>, <td>

-74.0
837.0
[<td>1551</td>, <td>2017-03-07</td>, <td>1630</td>, <td>328</td>, <td>11</td>, <td>-77.0</td>, <td>838.0</td>]
1551
2017-03-07
1630
328
11
-77.0
838.0
[<td>1550</td>, <td>2017-03-08</td>, <td>1631</td>, <td>329</td>, <td>11</td>, <td>-77.0</td>, <td>837.0</td>]
1550
2017-03-08
1631
329
11
-77.0
837.0
[<td>1552</td>, <td>2017-03-09</td>, <td>1632</td>, <td>330</td>, <td>12</td>, <td>-72.0</td>, <td>836.0</td>]
1552
2017-03-09
1632
330
12
-72.0
836.0
[<td>1554</td>, <td>2017-03-10</td>, <td>1633</td>, <td>330</td>, <td>12</td>, <td>-75.0</td>, <td>834.0</td>]
1554
2017-03-10
1633
330
12
-75.0
834.0
[<td>1553</td>, <td>2017-03-12</td>, <td>1634</td>, <td>331</td>, <td>12</td>, <td>-72.0</td>, <td>834.0</td>]
1553
2017-03-12
1634
331
12
-72.0
834.0
[<td>1555</td>, <td>2017-03-13</td>, <td>1635</td>, <td>331</td>, <td>12</td>, <td>-73.0</td>, <td>833.0</td>]
1555
2017-03-13
1635
331
12
-73.0
833.0
[<td>1556</td>, <td>2017-03-14</td>, <td>1636</td>, <td>332</td>, <td>12</td>, <td

1612
2017-05-11
1693
2
1
-76.0
840.0
[<td>1613</td>, <td>2017-05-12</td>, <td>1694</td>, <td>3</td>, <td>1</td>, <td>-76.0</td>, <td>840.0</td>]
1613
2017-05-12
1694
3
1
-76.0
840.0
[<td>1611</td>, <td>2017-05-13</td>, <td>1695</td>, <td>3</td>, <td>1</td>, <td>-77.0</td>, <td>840.0</td>]
1611
2017-05-13
1695
3
1
-77.0
840.0
[<td>1615</td>, <td>2017-05-14</td>, <td>1696</td>, <td>4</td>, <td>1</td>, <td>-75.0</td>, <td>841.0</td>]
1615
2017-05-14
1696
4
1
-75.0
841.0
[<td>1616</td>, <td>2017-05-15</td>, <td>1697</td>, <td>4</td>, <td>1</td>, <td>-77.0</td>, <td>841.0</td>]
1616
2017-05-15
1697
4
1
-77.0
841.0
[<td>1618</td>, <td>2017-05-16</td>, <td>1698</td>, <td>5</td>, <td>1</td>, <td>-77.0</td>, <td>842.0</td>]
1618
2017-05-16
1698
5
1
-77.0
842.0
[<td>1617</td>, <td>2017-05-17</td>, <td>1699</td>, <td>5</td>, <td>1</td>, <td>-76.0</td>, <td>843.0</td>]
1617
2017-05-17
1699
5
1
-76.0
843.0
[<td>1619</td>, <td>2017-05-18</td>, <td>1700</td>, <td>6</td>, <td>1</td>, <td>-76.0</td>, <

1678
2017-07-18
1759
34
2
-77.0
871.0
[<td>1679</td>, <td>2017-07-19</td>, <td>1760</td>, <td>35</td>, <td>2</td>, <td>-78.0</td>, <td>872.0</td>]
1679
2017-07-19
1760
35
2
-78.0
872.0
[<td>1684</td>, <td>2017-07-20</td>, <td>1761</td>, <td>35</td>, <td>2</td>, <td>-77.0</td>, <td>872.0</td>]
1684
2017-07-20
1761
35
2
-77.0
872.0
[<td>1687</td>, <td>2017-07-21</td>, <td>1762</td>, <td>36</td>, <td>2</td>, <td>-77.0</td>, <td>872.0</td>]
1687
2017-07-21
1762
36
2
-77.0
872.0
[<td>1680</td>, <td>2017-07-22</td>, <td>1763</td>, <td>36</td>, <td>2</td>, <td>-78.0</td>, <td>873.0</td>]
1680
2017-07-22
1763
36
2
-78.0
873.0
[<td>1682</td>, <td>2017-07-23</td>, <td>1764</td>, <td>37</td>, <td>2</td>, <td>-79.0</td>, <td>872.0</td>]
1682
2017-07-23
1764
37
2
-79.0
872.0
[<td>1690</td>, <td>2017-07-24</td>, <td>1765</td>, <td>37</td>, <td>2</td>, <td>-77.0</td>, <td>873.0</td>]
1690
2017-07-24
1765
37
2
-77.0
873.0
[<td>1688</td>, <td>2017-07-25</td>, <td>1766</td>, <td>38</td>, <td>2</td>, <td

2017-09-17
1818
61
3
-81.0
881.0
[<td>1737</td>, <td>2017-09-18</td>, <td>1819</td>, <td>62</td>, <td>3</td>, <td>-82.0</td>, <td>880.0</td>]
1737
2017-09-18
1819
62
3
-82.0
880.0
[<td>1738</td>, <td>2017-09-19</td>, <td>1820</td>, <td>62</td>, <td>3</td>, <td>-79.0</td>, <td>881.0</td>]
1738
2017-09-19
1820
62
3
-79.0
881.0
[<td>1740</td>, <td>2017-09-20</td>, <td>1821</td>, <td>63</td>, <td>3</td>, <td>-81.0</td>, <td>882.0</td>]
1740
2017-09-20
1821
63
3
-81.0
882.0
[<td>1739</td>, <td>2017-09-21</td>, <td>1822</td>, <td>63</td>, <td>3</td>, <td>-78.0</td>, <td>881.0</td>]
1739
2017-09-21
1822
63
3
-78.0
881.0
[<td>1743</td>, <td>2017-09-22</td>, <td>1823</td>, <td>64</td>, <td>3</td>, <td>-80.0</td>, <td>879.0</td>]
1743
2017-09-22
1823
64
3
-80.0
879.0
[<td>1742</td>, <td>2017-09-23</td>, <td>1824</td>, <td>64</td>, <td>3</td>, <td>-80.0</td>, <td>879.0</td>]
1742
2017-09-23
1824
64
3
-80.0
879.0
[<td>1741</td>, <td>2017-09-24</td>, <td>1825</td>, <td>64</td>, <td>3</td>, <td>-80.

-81.0
806.0
[<td>1820</td>, <td>2017-12-12</td>, <td>1902</td>, <td>99</td>, <td>4</td>, <td>-79.0</td>, <td>804.0</td>]
1820
2017-12-12
1902
99
4
-79.0
804.0
[<td>1821</td>, <td>2017-12-13</td>, <td>1903</td>, <td>100</td>, <td>4</td>, <td>-78.0</td>, <td>803.0</td>]
1821
2017-12-13
1903
100
4
-78.0
803.0
[<td>1822</td>, <td>2017-12-14</td>, <td>1904</td>, <td>100</td>, <td>4</td>, <td>-79.0</td>, <td>801.0</td>]
1822
2017-12-14
1904
100
4
-79.0
801.0
[<td>1823</td>, <td>2017-12-15</td>, <td>1905</td>, <td>101</td>, <td>4</td>, <td>-80.0</td>, <td>800.0</td>]
1823
2017-12-15
1905
101
4
-80.0
800.0
[<td>1824</td>, <td>2017-12-16</td>, <td>1906</td>, <td>101</td>, <td>4</td>, <td>-79.0</td>, <td>799.0</td>]
1824
2017-12-16
1906
101
4
-79.0
799.0
[<td>1825</td>, <td>2017-12-17</td>, <td>1907</td>, <td>101</td>, <td>4</td>, <td>-79.0</td>, <td>798.0</td>]
1825
2017-12-17
1907
101
4
-79.0
798.0
[<td>1826</td>, <td>2017-12-18</td>, <td>1908</td>, <td>102</td>, <td>4</td>, <td>-79.0</td>, <t

### Step 3: Store the Data

Assemble the scraped data into a Pandas DataFrame. The columns should have the same headings as the table on the website. Here’s an explanation of the column headings:

* `id`: the identification number of a single transmission from the Curiosity rover
* `terrestrial_date`: the date on Earth
* `sol`: the number of elapsed sols (Martian days) since Curiosity landed on Mars
* `ls`: the solar longitude
* `month`: the Martian month
* `min_temp`: the minimum temperature, in Celsius, of a single Martian day (sol)
* `pressure`: The atmospheric pressure at Curiosity's location

In [27]:
# Create an empty list
mars_weather = []
list_rows = html_soup.find_all('tr')
# Loop through the scraped data to create a list of rows
for tr in list_rows:
    column_heading = tr.find('th').text
    row_data = tr.find('td').text
    mars_weather[column_heading]=row_data
        

In [28]:
mars_weather

{}

In [ ]:
# Create a Pandas DataFrame by using the list of rows and a list of the column names


In [ ]:
# Confirm DataFrame was created successfully


### Step 4: Prepare Data for Analysis

Examine the data types that are currently associated with each column. If necessary, cast (or convert) the data to the appropriate `datetime`, `int`, or `float` data types.

  > **Hint** You can use the Pandas `astype` and `to_datetime` methods to accomplish this task.


In [ ]:
# Examine data type of each column


In [ ]:
# Change data types for data analysis


In [ ]:
# Confirm type changes were successful by examining data types again


### Step 5: Analyze the Data

Analyze your dataset by using Pandas functions to answer the following questions:

1. How many months exist on Mars?
2. How many Martian (and not Earth) days worth of data exist in the scraped dataset?
3. What are the coldest and the warmest months on Mars (at the location of Curiosity)? To answer this question:
    * Find the average the minimum daily temperature for all of the months.
    * Plot the results as a bar chart.
4. Which months have the lowest and the highest atmospheric pressure on Mars? To answer this question:
    * Find the average the daily atmospheric pressure of all the months.
    * Plot the results as a bar chart.
5. About how many terrestrial (Earth) days exist in a Martian year? To answer this question:
    * Consider how many days elapse on Earth in the time that Mars circles the Sun once.
    * Visually estimate the result by plotting the daily minimum temperature.


In [ ]:
# 1. How many months are there on Mars?


In [ ]:
# 2. How many Martian days' worth of data are there?


In [ ]:
# 3. What is the average low temperature by month?


In [ ]:
# Plot the average temperature by month


In [ ]:
# Identify the coldest and hottest months in Curiosity's location


In [ ]:
# 4. Average pressure by Martian month


In [ ]:
# Plot the average pressure by month


In [ ]:
# 5. How many terrestrial (earth) days are there in a Martian year?


On average, the third month has the coldest minimum temperature on Mars, and the eighth month is the warmest. But it is always very cold there in human terms!



Atmospheric pressure is, on average, lowest in the sixth month and highest in the ninth.

The distance from peak to peak is roughly 1425-750, or 675 days. A year on Mars appears to be about 675 days from the plot. Internet search confirms that a Mars year is equivalent to 687 earth days.

### Step 6: Save the Data

Export the DataFrame to a CSV file.

In [ ]:
# Write the data to a CSV


In [ ]:
browser.quit()